test/object-store/sync/client_reset.cpp

```diff
@@ -5364,23 +5424,18 @@ TEST_CASE("client reset with nested collection", "[client reset][local][nested c
                     REQUIRE(nlist.get_any(8).get_int() == 7);
                 }
                 else {
+                    // db must be equal to local
                     List list{local_realm, obj, col};
                     REQUIRE(list.size() == 1);
                     auto nlist = list.get_list(0);
-                    REQUIRE(nlist.size() == 13);
+                    REQUIRE(nlist.size() == 7);
                     REQUIRE(nlist.get_any(0).get_int() == 1);
                     REQUIRE(nlist.get_any(1).get_int() == 2);
                     REQUIRE(nlist.get_any(2).get_int() == 3);
                     REQUIRE(nlist.get_any(3).get_int() == 4);
                     REQUIRE(nlist.get_any(4).get_int() == 5);
                     REQUIRE(nlist.get_any(5).get_int() == 6);
                     REQUIRE(nlist.get_any(6).get_int() == 7);
-                    REQUIRE(nlist.get_any(7).get_int() == 4);
-                    REQUIRE(nlist.get_any(8).get_int() == 5);
-                    REQUIRE(nlist.get_any(9).get_int() == 8);
-                    REQUIRE(nlist.get_any(10).get_int() == 9);
-                    REQUIRE(nlist.get_any(11).get_int() == 6);
-                    REQUIRE(nlist.get_any(12).get_int() == 7);
```

Comment:
There are pros and cons to each outcome. But this change sits better with me because it is consistent with how top level lists are recovered. 👍